pkg/client/interface.go

```diff
@@ -313,3 +314,20 @@ type SharedQueryClient interface {
 type BlockQueryClient interface {
 	Block(ctx context.Context, height *int64) (*coretypes.ResultBlock, error)
 }
+
+// ProofParams is a go interface type which corresponds to the poktroll.proof.Params
+// protobuf message. Since the generated go types don't include interface types, this
+// is necessary to prevent dependency cycles.
+type ProofParams interface {
```

Comment:
**Question**: I understand why we did this but just have to ask if there's no way around it?

**Thinking**: My aversion to stuff like this is needing to maintain multiple source of truths.

**Idea**:  What if we created an `expected_interfaces.go` or `dependent_interfaces` or `upstream` to distinguish interfaces we're defining or using.

Reply:
Any alternative would involve some kind of re-organization of packages.

Reply:
For now, I'll move it to an `expected_interfaces.go` in the same package.